### PR TITLE
feat(feudal): rewrite feudal send script to python

### PR DESCRIPTION
**Pull request by xflord**

### CODE REVIEW

**Review by mattjoke — APPROVED** [review]
Looks good, if tested👍

File: send/feudal.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
+#!/usr/bin/env python3.7
```
Comment (balcirakpeter) [inline send/feudal.py:1]:
```suggestion
#!/usr/bin/env python3
```

File: send/feudal.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+if re.fullmatch(re.compile(r'.*\.\..*'), credentials_filename):
+	send_lib.die_with_error("Unsupported destination (\"..\" cannot be part of destination name)")
+
+with open(f'/etc/perun/services/feudal/{credentials_filename}', 'r') as f:
+	credentials = f.read().strip()
+
+if not credentials:
+	send_lib.die_with_error('No credentials available in configuration')
+# TODO verify credentials file format
+auth = (credentials.split(' ')[0], credentials.split(' ')[1])
```
Comment (balcirakpeter) [inline send/feudal.py:34]:
```suggestion
auth = (credentials.split(':')[0], credentials.split(':')[1])
```

File: send/feudal.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+service_files_dir = send_lib.get_gen_folder(facility, SERVICE_NAME)
+
+send_lib.create_lock(SERVICE_NAME, destination)
```
Comment (balcirakpeter) [inline send/feudal.py:38]: This does not work as the destination contains `//`. Overall we need to escape such characters in the `send_lib.py` so we do not encounter the same problem next time. Can you please add the escaping there?

**Review by balcirakpeter — CHANGES_REQUESTED** [review]

File: send/feudal.py (hunk 3 above)
Comment (balcirakpeter) [inline send/feudal.py:38]:
```suggestion
send_lib.create_lock(SERVICE_NAME, re.sub(r'^https?://', '', destination))
```

**Pull request merged**